### DISCUSSION TITLE
dynamic class attached to a click event keeps getting triggered

### DISCUSSION THREAD

**Discussion by eduardonwa** [alpinejs/alpine#3636] [1. Help]
on my navbar, only if a category has subcategories a chevron icon will appear next to it, making a list of subcategories be displayed.
this is controlled by the "click" event for mobile and for desktop its hover.

the problem is that on mobile devices the event keeps getting fired too many times making it impossible to open the subcategories menu or even going to that category main page. 

```
    <div 
        x-data="{ open: null }" 
        class="w-full flex-grow sm:flex sm:items-center sm:w-auto"
    >
        <div class="mx-auto flex flex-col items-center justify-center sm:flex-row text-sm font-bold uppercase mt-0 px-6 py-2">
        @foreach ($allCategories as $category)
                <div 
                    class="py-2 relative w-full lg:w-auto" 
                    @mouseenter="open = '{{ $loop->index }}'" 
                    @mouseleave="open = null"
                >
                    <div class="flex items-center justify-between md:flex-none">
                        <a 
                            class="text-white text-lg transition ease-out hover:bg-gray-100 hover:text-black rounded py-2 px-4 mx-2"
                            href="{{ route('by-category', ['category'=>$category['slug']]) }}" 
                        >
                            {{ $category['title'] }} 
                        </a>
                        @if ($category['parent_id'] == null && $category->subCategory()->count() > 0)
                        <i :class="open === '{{ $loop->index }}' ? 'fa-chevron-up': 'fa-chevron-down'" 
                            class="fas ml-2 text-white font-bold cursor-pointer text-lg"
                            @click="open = open === '{{ $loop->index }}' ? null : '{{ $loop->index }}'">
                        </i>
```

**Comment by ekwoka** [comment]
are you actually testing on mobile or just mobile emulation in the browser?

But I believe it's because the mouseenter event is still triggered by touches.

That's just how the spec is

https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseenter_event

note the use of `pointer` as the thing, not simply a mouse.

What you probably want is to do this with more css than js.

**Reply by ekwoka** [reply]
That's not really a helpful question

**Reply by eduardonwa** [reply]
i am still testing on mobile, sorry i wasn't able to understand what you meant but i think im kind of getting it now. i am able to show the chevron and the subcategories menu, but maybe this is where i wasn't good at exaplaining my problem, if i try to click on one of the sub categories it just won't work. when i looked to inspect the element i saw that the event would get triggered non-stop. 

that's basically the issue, again sorry for not explaining it that well hehe. if you have time to test this yourself, try visiting the site @ sickofmetal.net on your mobile device, thanks.

**Reply by ekwoka** [reply]
which event gets triggered nonstop?

**Reply by eduardonwa** [reply]
the mouse enter

**Reply by ekwoka** [reply]
Is it possible that the active state moves the element away fromt he point and vice versa?

So the element is moving back and forth under the pointer?

**Reply by eduardonwa** [reply]
when i tap on the item nothing happens i thought this was because there wasn't enough space for the touch event to actually trigger an event, but i checked for padding and there is enough. 

i think i could find another way of triggering the dynamic classes, maybe.

but to answer your question i don't know how to debug it so i don't really know. by the way this bug its only on mobile, firefox mobile viewport "works" but because i am using the mouse, like i said the website is live and when you tap on some menu item nothing is gonna happen

edit
i was replacing the font awesome classes to something different, and i think these are causing the excessive triggering, yep i just tested this on the live version i removed those classes and now those links work

**Reply by ekwoka** [reply]
Well, FA is pretty awful, so that's good 😄 

Probably had something to do with it changing the image causing reflows that moved the bounding box off.
